### DISCUSSION TITLE
[PR review] Allow `viewed` checkbox on commit per commit review

### DISCUSSION THREAD

**Discussion by mpetitdant** [community/community#5303] [Pull Requests]
In our org, delivering a feature sometimes leads to large PRs to bring value. Our devs use to split their work by commit and suggest a commit per commit review. But in that case the `viewed` button is not present. 

I'm aware that it is possible that my review on the first commit may be irrelevant if a following commit changes the code again, but even in that case it would help me as I would not have to remember which files I viewed or not. Especially when I need to review a story in a non linear way (either the developper's way to do commits, or the GH way to list files).

**Comment by mpetitdant** [comment]
My issues is a bit related to https://github.com/github/feedback/discussions/5054

An in-between solution may be to allow individual per-commit reviews, done by one or many reviewers, and mark the PR as reviewed only when all commits are individually reviewed.

**Reply by iandunn** [reply]
Related: #6089

**Comment by LuigiCardamone** [comment]
This feature would be very helpful. It was also reported here: https://github.com/isaacs/github/issues/1635

**Comment by BenMorel** [comment]
This would be particularly useful when reviewing open source pull requests, as well.
I'm using this feature daily on GitLab and miss it here!

**Comment by bweisberger** [comment]
Agreed, this would make my life easier

**Comment by thockin** [comment]
There seems to be no rhyme or reason as to when the "viewed" button is available and when it is not.  I thought maybe it was because a file was touched in multiple commits, but I am looking at a PR with lots of commits.  I chose a file which is only touched in ONE SINGLE commit, and it doesn't have the button, while other files do.

Commit-by-commit is such a useful pattern for review, but it's totally second-class in the UI.
